### PR TITLE
Feature/cart page

### PR DESCRIPTION
**Pull request by HIMDanny**
Add cart page layout

Please, notice that this is only jsx layout and it doesn't include functionality that will be implemented in future

### CODE REVIEW

**Comment by RomanRymarchuk1** [comment]
Снимок экрана 2022-11-13 в 18 01 11

I think it's better to use ul and li tags here

**Comment by RomanRymarchuk1** [comment]
Снимок экрана 2022-11-13 в 18 28 14
 
also it can be section tag

**Comment by RomanRymarchuk1** [comment]
Снимок экрана 2022-11-13 в 18 39 47

maybe it would be better to center the buttons

**Comment by RomanRymarchuk1** [comment]
Снимок экрана 2022-11-13 в 18 42 17

you don't have props in this component
did you add it for the future?

**Comment by HIMDanny** [comment]
> Снимок экрана 2022-11-13 в 18 42 17
> 
> you don't have props in this component did you add it for the future?

**Comment by HIMDanny** [comment]
> Снимок экрана 2022-11-13 в 18 42 17
> 
> you don't have props in this component did you add it for the future?

Yes, there will be props that contain tour data

**Comment by HIMDanny** [comment]
> Снимок экрана 2022-11-13 в 18 01 11
> 
> I think it's better to use ul and li tags here

No problem. Changed

**Comment by HIMDanny** [comment]
> Снимок экрана 2022-11-13 в 18 28 14
> 
> also it can be section tag

Done

**Comment by HIMDanny** [comment]
> 

Don't think that it is an option as it is a position of buttons by default in dialog, so I don't see the reason to change this behavior.
Maybe you can give me arguments about that?

**Comment by HIMDanny** [comment]
> Снимок экрана 2022-11-13 в 18 39 47
> 
> maybe it would be better to center the buttons

Don't think that it is an option as it is a position of buttons by default in dialog, so I don't see the reason to change this behavior.
Maybe you can give me arguments about that?

**Comment by RomanRymarchuk1** [comment]
this is my subjective opinion, center positioning looks nicer
if you don't think so, you can leave it like that
but in any case, I should suggest adding an indent from the bottom of the buttons the same as the top of the title

**Comment by HIMDanny** [comment]
> this is my subjective opinion, center positioning looks nicer if you don't think so, you can leave it like that but in any case, I should suggest adding an indent from the bottom of the buttons the same as the top of the title

I've increased bottom indentation.

In my humble opinion, default position looks better)

image
image

**Comment by RomanRymarchuk1** [comment]
good job!)

**Review by RomanRymarchuk1 — APPROVED** [review]

File: client/src/features/Cart/pages/CartPage/CartPage.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+import { Box, Container, Stack, styled, Typography } from '@mui/material';
+import { CartItem } from '../../components';
+import TotalInfoDialog from '../../components/TotalInfoDialog/TotalInfoDialog';
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/features/Cart/pages/CartPage/CartPage.js]: why didnt you export this component from the index.js file created inside the components folder of this feature?

File: client/src/features/Cart/pages/CartPage/CartPage.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  gap: theme.spacing(2),
+
+  listStyleType: 'none',
+}));
+
+function CartPage() {
+  return (
+    <Box marginTop="80px" marginBottom="40px" component="section">
+      <Container>
+        <Typography variant="h1">Cart</Typography>
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/features/Cart/pages/CartPage/CartPage.js]:
```suggestion
        <Typography variant="h2">Cart</Typography>
```

File: client/src/features/Cart/components/TotalInfoDialog/TotalInfoDialog.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        <Stack direction="row" justifyContent="space-between">
+          <Typography>To be paid:</Typography>
+          <Typography fontSize={18}>2000 ₴</Typography>
+        </Stack>
+      </Section>
+
+      <MuiBox>
+        <Stack direction="row" justifyContent="end" alignItems="center">
+          <Button sx={{ paddingInline: '30px' }} disableElevation onClick={() => navigate('/checkout')}>
+            Make an order
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/features/Cart/components/TotalInfoDialog/TotalInfoDialog.js]:
```suggestion
            Checkout
```

File: client/src/features/Cart/components/TotalInfoDialog/TotalInfoDialog.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+import { Button, Stack, Box as MuiBox, Typography, styled } from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+
+const BoxWrapper = styled(MuiBox)(({ theme }) => ({
+  // ? should it be sticky or make it stay on its place?
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/features/Cart/components/TotalInfoDialog/TotalInfoDialog.js]: love the idea of it being sticky.

File: client/src/features/Cart/components/CartItem/CartItem.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+  Typography,
+  CardContent as MuiCardContent,
+  IconButton,
+  TextField,
+} from '@mui/material';
+
+import { AccessTime, Remove, Add, DeleteOutline } from '@mui/icons-material';
+import { ReactComponent as CoinsIcon } from '../../../../svg/CoinsIcon.svg';
+
+import DeleteItemModal from '../DeleteItemModal/DeleteItemModal';
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/features/Cart/components/CartItem/CartItem.js]:
```suggestion
import { DeleteItemModal } from '../index';
```

File: client/src/features/Cart/pages/CartPage/CartPage.js (hunk 1 above)
Comment (HIMDanny) [reply client/src/features/Cart/pages/CartPage/CartPage.js]: make sense) My fault. Fixed

File: client/src/features/Cart/pages/CartPage/CartPage.js (hunk 2 above)
Comment (HIMDanny) [reply client/src/features/Cart/pages/CartPage/CartPage.js]: Done

File: client/src/features/Cart/components/TotalInfoDialog/TotalInfoDialog.js (hunk 3 above)
Comment (HIMDanny) [reply client/src/features/Cart/components/TotalInfoDialog/TotalInfoDialog.js]: Done

File: client/src/features/Cart/components/CartItem/CartItem.js (hunk 5 above)
Comment (HIMDanny) [reply client/src/features/Cart/components/CartItem/CartItem.js]: Done

**Review by Ahmed-Fahmy-Dev — APPROVED** [review]
will approve but please be informed that tests are missing

**Pull request merged**